common/interest-confirmation/tutoring/receiver/people/db-queries.ts

```diff
@@ -0,0 +1,56 @@
+import { EntityManager, SelectQueryBuilder } from "typeorm";
+import { matchableTuteesQuery } from "../../../../administration/match-making/tutoring/people/tutees";
+import { RegistrationSource } from "../../../../entity/Person";
+import { Pupil } from "../../../../entity/Pupil";
+import { InterestConfirmationStatus } from "../../../../entity/PupilTutoringInterestConfirmationRequest";
+
+// Raw Queries:
+// ⎺⎺⎺⎺⎺⎺⎺⎺
+function allMatchablePupilsThatRequireInterestConfirmationQuery(manager: EntityManager) {
```

Comment:
@geroembser could you please take the time to clean this up? It took us about 1 hour to figure out what the hell this code does. I also doubt that having one function for each step improves readability.

Reply:
I don’t see why. As I’ve said, this was just a temporary solution. As far as I know this interest confirmation system will be completely replaced by a more sophisticated _match_ confirmation system in the very near future. (actually it should have been replaced a few weeks ago, but that's another story...)
If you have an idea how to clean this up easily – I’m open to that. For example, rewriting that stuff with Prisma would definitely help. As this would solve the problem why I have introduced that separate functions: working with the queries being pure strings and have no checks for the underlying model (until runtime) made it difficult to develop and test since I had to change this string in different places (and that annoyed because I’m really not familiar with typeorm's raw sql query syntax). 
So I really don't see why I should bother at this point. Then I would directly implement the new match confirmation system. The "specification" process here should be completed now… (but I guess it hasn't left Slack until now...)

Reply:
Ah, didn't know that. I'm fine leaving it as is for now if it is replaced anyway in the (hopefully) near future.